src/components/ToggleSwitch/index.tsx

```diff
@@ -0,0 +1,210 @@
+    top: dimen.normalizeDimen(2),
+    right: dimen.normalizeDimen(3)
+  },
+  mainStyes: {
+    borderRadius: dimen.normalizeDimen(12),
+    backgroundColor: '#F5F5F5',
+    height: dimen.normalizeDimen(20),
+    width: dimen.normalizeDimen(42),
+    padding: dimen.normalizeDimen(2),
+    ...Platform.select({
+      ios: {
+        shadowColor: 'rgba(0,0,0,0.25)',
+        shadowOffset: {width: 0, height: 0},
+        shadowOpacity: dimen.normalizeDimen(1),
+        shadowRadius: dimen.normalizeDimen(1)
+      },
+      android: {
+        elevation: dimen.normalizeDimen(1)
+      }
+    })
+  },
+  labelLeft: {
+    marginRight: 5,
+    color: '#FFFFFF',
+    fontSize: 12,
+    fontWeight: '400'
+  },
+  labelRight: {
+    marginLeft: 5,
+    color: '#FFFFFF',
+    fontSize: 12,
+    fontWeight: '400'
+  }
+});
+
+interface ToggleSwitchProps {
+  value?: boolean;
+  labelLeft?: string;
+  labelRight?: string;
+  circleActiveColor?: string;
+  circleInActiveColor?: string;
+  backgroundActive?: string;
+  backgroundInactive?: string;
+  styleLabelLeft?: StyleProp<TextStyle>;
+  styleLabelRight?: StyleProp<TextStyle>;
+  activeTextColor?: string;
+  inactiveTextColor?: string;
+  containerStyle?: TouchableOpacityProps['style'];
+  labelOff?: string;
+  labelOn?: string;
+  onValueChange?: (value: boolean) => void;
+}
+
+const ToggleSwitch: React.FC<ToggleSwitchProps> = ({
+  value = false,
+  onValueChange,
+  labelLeft,
+  labelRight,
+  circleActiveColor = '#00ADB5',
+  circleInActiveColor = '#2C67BC',
+  backgroundActive = '#F5F5F5',
+  backgroundInactive = '#F5F5F5',
+  styleLabelLeft,
+  styleLabelRight,
+  activeTextColor = '#00ADB5',
+  inactiveTextColor = '#2C67BC',
+  containerStyle,
+  labelOff = 'Off',
+  labelOn = 'On'
+}) => {
+  const positionButton = React.useMemo(() => new Animated.Value(0), []);
+
+  const startAnimToOff = () => {
+    Animated.timing(positionButton, {
+      toValue: 0,
+      duration: 150,
+      easing: Easing.ease,
+      useNativeDriver: false
+    }).start();
+  };
+
+  const startAnimToOn = () => {
+    Animated.timing(positionButton, {
+      toValue: 1,
+      duration: 150,
+      easing: Easing.ease,
+      useNativeDriver: false
+    }).start();
+  };
+
+  const positionInterPol = positionButton.interpolate({
+    inputRange: [0, 1],
+    outputRange: [0, dimen.normalizeDimen(22)]
+  });
+  const backgroundColorAnim = positionButton.interpolate({
+    inputRange: [0, 1],
+    outputRange: [backgroundInactive, backgroundActive]
+  });
+
+  const initialOpacityOn = positionButton.interpolate({inputRange: [0, 1], outputRange: [0, 1]});
+  const initialOpacityOff = positionButton.interpolate({inputRange: [0, 1], outputRange: [1, 0]});
+
+  const onPress = () => {
+    if (!value) {
+      startAnimToOff();
+    } else {
+      startAnimToOn();
+    }
+    if (onValueChange) onValueChange(!value);
+  };
+
+  if (value) {
+    startAnimToOn();
+  } else {
+    startAnimToOff();
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The animation start functions are called directly in the component body. This could lead to unnecessary animations when the component re-renders. Consider moving these calls into a `useEffect` hook that only runs when the `value` prop changes.

```diff
+ React.useEffect(() => {
+   if (value) {
+     startAnimToOn(positionButton);
+   } else {
+     startAnimToOff(positionButton);
+   }
+ }, [value]);

- if (value) {
-   startAnimToOn();
- } else {
-   startAnimToOff();
- }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->